engine/src/main/java/org/terasology/utilities/modifiable/ModifiableValue.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.utilities.modifiable;
+
+/**
+ * A helper type to get and modify the value of a component without changing its actual value
+ * <p>
+ * The result value is guaranteed to be greater or equal to zero. Components using this type must mention so in their
+ * javadoc.
+ * </p>
+ */
+public class ModifiableValue {
+    private final float baseValue;
+
+    private float preModifiers;
+    private float multipliers;
+    private float postModifiers;
+
+    public ModifiableValue(float baseValue) {
+        this.baseValue = baseValue;
+    }
+
+    public float getBaseValue() {
+        return baseValue;
+    }
+
+    public ModifiableValue multiply(float amount) {
+        multipliers += amount;
+        return this;
+    }
+
+    public ModifiableValue preAdd(float amount) {
+        preModifiers += amount;
+        return this;
+    }
+
+    public ModifiableValue postAdd(float amount) {
+        postModifiers += amount;
+        return this;
+    }
+
+    /**
+     * Calculates the result value from the base value and given modifiers and multipliers.
+     * <p>
+     * The value is calculated based on the following formula:
+     * <pre>
+     * result = max(0, <baseValue> + Σ <modifier> * Π <multiplier> + Σ <postModifier>)
```

Comment:
This formula should be updated to match the actual value: `result = (<baseValue> + Σ <modifier>) * Σ <multiplier> + Σ <postModifier>`

Reply:
Have changed the multiplers to compound now as per the formula

Reply:
The rest of the formula in the comment still needs changing though: `result = (<baseValue> + Σ <modifier>) * Π <multiplier> + Σ <postModifier>`